### ISSUE TITLE
Cannot use az login to get my subscriptions

### ISSUE THREAD

**Issue by rokeno** [Azure/azure-cli#11806]
## Describe the bug

**Command Name**
`az login`

**Errors:**

:open_mouth:

```The command failed with an unexpected error. Here is the traceback:

Error occurred in request., SSLError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\contrib\pyopenssl.py", line 453, in wrap_socket
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\pyOpenSSL\OpenSSL\SSL.py", line 1915, in do_handshake
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\pyOpenSSL\OpenSSL\SSL.py", line 1647, in _raise_ssl_error
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\pyOpenSSL\OpenSSL\_util.py", line 54, in exception_from_error_queue
OpenSSL.SSL.Error: [('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 600, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 343, in _make_request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 839, in _validate_conn
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connection.py", line 344, in connect
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\util\ssl_.py", line 344, in ssl_wrap_socket
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\contrib\pyopenssl.py", line 459, in wrap_socket
ssl.SSLError: ("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\adapters.py", line 449, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 667, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 667, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 667, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\connectionpool.py", line 638, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\urllib3\urllib3\util\retry.py", line 398, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\universal_http\requests.py", line 137, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\sessions.py", line 533, in request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\sessions.py", line 646, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\adapters.py", line 514, in send
requests.exceptions.SSLError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 326, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 384, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 377, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 354, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\commands\__init__.py", line 145, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\__init__.py", line 451, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-profile\azure\cli\command_modules\profile\custom.py", line 133, in login
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 190, in find_subscriptions_on_login
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 787, in find_through_authorization_code_flow
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-cli-core\azure\cli\core\_profile.py", line 831, in _find_using_common_tenant
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\paging.py", line 143, in __next__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\paging.py", line 129, in advance_page
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\azure-mgmt-resource\azure\mgmt\resource\subscriptions\v2016_06_01\operations\tenants_operations.py", line 80, in internal_paging
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\service_client.py", line 336, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\pipeline\__init__.py", line 197, in run
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\pipeline\__init__.py", line 150, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\pipeline\requests.py", line 72, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\pipeline\requests.py", line 137, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\pipeline\__init__.py", line 150, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\pipeline\requests.py", line 193, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\universal_http\requests.py", line 328, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\universal_http\requests.py", line 140, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\exceptions.py", line 51, in raise_with_traceback
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\msrest\msrest\universal_http\requests.py", line 137, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\sessions.py", line 533, in request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\sessions.py", line 646, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-z6u9vbmn\requests\requests\adapters.py", line 514, in send
msrest.exceptions.ClientRequestError: Error occurred in request., SSLError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))
```

## To Reproduce:
Steps to reproduce the behavior. Note: Command arguments have been redacted.

- `Fill in additional info here`
- `Run: az login`

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli                         2.0.62

```

**Comment by jsntcy** [comment]
@jiasli , please help take a look.

**Comment by yonzhan** [comment]
add to S165.

**Comment by yonzhan** [comment]
@qianwens  you can refer to https://github.com/Azure/azure-cli/pull/11093, try to repro locally and fix this issue.

**Comment by qianwens** [comment]
@rokeno, error message "certificate verify failed" may throw when there is a proxy in your network, you can follow this doc https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#working-behind-a-proxy to fix. The error message improvement has been fixed in [#11093](https://github.com/Azure/azure-cli/pull/11093), you can upgrade Az CLI to the latest version and have a try.

**Comment by rokeno** [comment]
This issue has been solved. I am sorry for the time spent.
I appreciate you have taken some for this, but this issue can now be closed. The results are being displayed when i am using az login (in powershell).
Thanks.

**Comment by inaun** [comment]
> @rokeno, error message "certificate verify failed" may throw when there is a proxy in your network, you can follow this doc https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#working-behind-a-proxy to fix. The error message improvement has been fixed in [#11093](https://github.com/Azure/azure-cli/pull/11093), you can upgrade Az CLI to the latest version and have a try.

Why does everyone want to keep pointing people to this poor article? It doesn't help resolve this problem.

**Comment by jiasli** [comment]
@inaun, the instruction provided in the article (now moved to [Work behind a proxy](https://docs.microsoft.com/en-us/cli/azure/use-cli-effectively#work-behind-a-proxy)) is the standard workflow for Python apps to work behind a proxy. We have tested and verified it multiple times and it did help lots of users solve their problems.

If you find it is not helpful and can't solve your problem, please kindly provide 

1. the error message you are seeing
2. the steps you have taken
3. your network configuration

Thanks for the information and understanding.
